pyOxygenSCPI/oxygenscpi.py

```diff
@@ -532,31 +579,162 @@ def setElogPeriod(self, period):
+            num_ch += 1
         #print(len(data)/(1.0*num_ch), data)
         num_items = int(len(data)/num_ch)
         data = [data[i*num_ch:i*num_ch+num_ch] for i in range(num_items)]
+        if not raw_string:
+            for i, row in enumerate(data):
+                data[i] = self._convertElogArray(row)
         return data
 
+    def _convertElogArray(self, data_array):
+        """Converts a single array from fetchElog string values into float.
+        
+        If the Elog timestamp is set to 'ABS' then the first value of the array
+        is converted into datetime object.
+
+        Args:
+        data_array : list of strings
+            List containing single array of string measurements from fetchElog.
+
+        Returns:
+            List with value types converted to float (datetime for value at
+            index 0 if timestamp is ABS).
+        """
+        # When ELOG timestamp is set to ABS, the first value of a row is a
+        # datetime string and the remaining values are strings that can
+        # be converted to floating point numbers.
+        if self.elogTimestamp == "ABS":
+            new_array = []
+            dtime = data_array[0].replace('"', '')
+            # Oxygen's datetime with whole seconds does not display microseconds
+            # for those cases the ".%f" formatting is excluded.
+            fmt = '%Y-%m-%dT%H:%M:%S.%f' if '.' in dtime else '%Y-%m-%dT%H:%M:%S'
+            new_array.append(dt.datetime.strptime(dtime, fmt))
+            new_array.extend(float(value) for value in data_array[1:])
+        else:
+            new_array = [float(value) for value in data_array]
+        return new_array
+
+    def fetchElogAccumulated(self, timeout=10):
+        """Fetch ELOG until the actual timestamp is reached.
+
+        This function blocks the execution and keeps fetching elog values until
+        a fetched timestamp is higher than the timestamp saved at the moment in
+        which the function was called. If called right after starting the
+        external logging, it is possible that the function needs longer because
+        the update of the Dewetron's buffer is not instantaneous.
+
+        Depending on the internet connection, the function itself can take a few
+        seconds to be excecuted.
+        
+        It requires the elog timestamp to be either 'ABS' or 'ELOG'.
+        - With 'ABS' timestamp, the stop condition will compare the absolute
+        timestamp from the system executing the function with the timestamp from
+        the operating system in which Oxygen is running.
+        - With 'ELOG' timestamp, the stop condition will be met when the fetched
+        timestamp added to the timestamp in which the startElog function was
+        called (tracked by _localElogStartTime) attribute is higher than the
+        timestamp from execution of the function.
+
+        Args:
+            timeout (float): timeout for the accumulated fetching.
```

Comment:
Please add the **unit of the timeout** (seconds), maybe also type-hint `timeout: float = 10` (type hinting is a goal with this library in the long-term).
I didn't check if there could be any problems with timezones (i.e. the python client is in a different zone than Oxygen). But it would be sufficient to handle this when problems arise.

Reply:
No problem, I did not see any type hints before, therefore I also did not include them. But now I added it for the `fetchElog`, `_convertElogArray`, and `fetchElogAccumulated`. The typing might seem complicated to read because I used `Union` instead of `|` so that the code works for Python versions < 3.10.

Regarding the timezone, I think it can be solved easily using the same approach from the method `_get_value_from_ascii`.